govcd/entity_supervisor_namespace_test.go

```diff
@@ -0,0 +1,254 @@
+			ClassName:   "small",
+			Description: check.TestName(),
+			InitialClassConfigOverrides: ccitypes.SupervisorNamespaceSpecInitialClassConfigOverrides{
+				StorageClasses: []ccitypes.SupervisorNamespaceSpecInitialClassConfigOverridesStorageClass{
+					{
+						Name:     storagePolicy,
+						LimitMiB: 256,
+					},
+				},
+				Zones: []ccitypes.SupervisorNamespaceSpecInitialClassConfigOverridesZone{
+					{
+						CpuLimitMHz:          200,
+						CpuReservationMHz:    1,
+						MemoryLimitMiB:       256,
+						MemoryReservationMiB: 1,
+						Name:                 supervisorZoneName,
+					},
+				},
+			},
+			RegionName: regionName,
+			VpcName:    vpcName,
+		},
+	}
+
+	newNsAddr, err := vcd.client.Client.GetEntityUrl(fmt.Sprintf(ccitypes.SupervisorNamespacesURL, newProject.Name))
+	check.Assert(err, IsNil)
+	check.Assert(newNsAddr, NotNil)
+
+	newNs := &ccitypes.SupervisorNamespace{}
+	// Create
+	err = vcd.client.Client.PostEntity(newNsAddr, nil, nsCfg, newNs, nil)
+	check.Assert(err, IsNil)
+	check.Assert(newNs, NotNil)
+	check.Assert(newNs.GenerateName, Equals, nsCfg.GenerateName)
+	check.Assert(strings.HasPrefix(newNs.Name, nsCfg.GenerateName), Equals, true)
+
+	// Get
+	nsAddr, err := vcd.client.Client.GetEntityUrl(fmt.Sprintf(ccitypes.SupervisorNamespacesURL, newProject.Name), "/", newNs.Name)
+	check.Assert(err, IsNil)
+	check.Assert(nsAddr, NotNil)
+
+	_, err = waitForEntityState(&vcd.client.Client, nsAddr, []string{"CREATING", "WAITING"}, []string{"CREATED"})
+	check.Assert(err, IsNil)
+
+	// defer namespace cleanup
+	defer func() {
+		p := &ccitypes.SupervisorNamespace{}
+		err = vcd.client.Client.GetEntity(nsAddr, nil, p, nil)
+		if err != nil && !strings.Contains(err.Error(), "404") {
+			err := vcd.client.Client.DeleteEntity(nsAddr, nil, nil)
+			check.Assert(err, IsNil)
+			_, err = waitForEntityState(&vcd.client.Client, nsAddr, []string{"DELETING", "WAITING"}, []string{"DELETED"})
+			check.Assert(err, IsNil)
+		}
+	}()
+
+	getNs := &ccitypes.SupervisorNamespace{}
+	err = vcd.client.Client.GetEntity(nsAddr, nil, getNs, nil)
+	check.Assert(err, IsNil)
+	check.Assert(getNs, NotNil)
+	check.Assert(getNs.Name, Equals, newNs.Name)
+
+	// Delete
+	err = vcd.client.Client.DeleteEntity(nsAddr, nil, nil)
+	check.Assert(err, IsNil)
+	_, err = waitForEntityState(&vcd.client.Client, nsAddr, []string{"DELETING", "WAITING"}, []string{"DELETED"})
+	check.Assert(err, IsNil)
+
+	// Check it is deleted
+	err = vcd.client.Client.GetEntity(nsAddr, nil, getProject, nil)
+	check.Assert(err, NotNil)
+	check.Assert(strings.Contains(err.Error(), "404"), Equals, true)
+
+	// Delete project
+	err = vcd.client.Client.DeleteEntity(projectAddr, nil, nil)
+	check.Assert(err, IsNil)
+
+	// Check it is deleted
+	err = vcd.client.Client.GetEntity(projectAddr, nil, getProject, nil)
+	check.Assert(strings.Contains(err.Error(), "404"), Equals, true)
+}
+
+func getAnyEntityState(client *Client, urlRef *url.URL) (*AnyEntityStatus, error) {
```

Comment:
The problem with this is that there are no standards regarding the status of a k8s resource; some of them have a `status` subresource, but others do not; some have `conditions` and `phase`, but others do not. It may be better to make the above functions specific for the supervisornamespace resource.

Reply:
Ok. Will change names. I explicitly put this into _test.go file so that it is not exposed anywhere and is used only for internal testing.

Reply:
Changed name and also got rid of some unused structure - it is now tailored for Namespaces.